xdp-dump/xdpdump.c

```diff
@@ -0,0 +1,628 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+
+/*****************************************************************************
+ * Include files
+ *****************************************************************************/
+#include <ctype.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <bpf/bpf.h>
+#include <bpf/libbpf.h>
+
+#include <linux/perf_event.h>
+
+#include <net/if.h>
+
+#define PCAP_DONT_INCLUDE_PCAP_BPF_H
+#include <pcap/dlt.h>
+#include <pcap/pcap.h>
+
+#include <sys/resource.h>
+#include <sys/sysinfo.h>
+
+#include "params.h"
+#include "logging.h"
+#include "util.h"
+#include "xdpdump.h"
+
+/*****************************************************************************
+ * Local definitions and global variables
+ *****************************************************************************/
+#define PROG_NAME "xdpdump"
+#define DEFAULT_SNAP_LEN 262144
+
+enum func_directions {
+	PROG_IN,
+	PROG_OUT,
+	PROG_INOUT
+};
+
+static const struct dumpopt {
+	bool                  list_interfaces;
+	struct iface          iface;
+	uint32_t              snaplen;
+	char                 *pcap_file;
+	enum func_directions  rx_capture;
+} defaults_dumpopt = {
+	.list_interfaces = false,
+	.snaplen = DEFAULT_SNAP_LEN,
+	.rx_capture = PROG_IN,
+};
+struct dumpopt cfg_dumpopt;
+
+struct enum_val func_direction_options[] = {
+	{"fentry", PROG_IN},
+	{"fexit", PROG_OUT},
+	{"fentryfexit", PROG_INOUT},
+	{"fexitfentry", PROG_INOUT},
```

Comment:
How about just making the direction an OPT_FLAGS entry? That way people can just specify fentry,fexit as separate options, and there'll be no need to duplicate them?

Reply:
Will change this to use OPT_FLAGS (note to self, update documentation :)

Reply:
Great! Should simplify some of the matching code as well...